tests/bluetooth/tester/src/btp_vcp.c

```diff
@@ -0,0 +1,587 @@
+		int err = bt_aics_gain_set(included.aics[0], gain);
+
+		if (err) {
+			tester_rsp(BTP_SERVICE_ID_AICS, AICS_SET_GAIN,
+				   CONTROLLER_INDEX, BTP_STATUS_FAILED);
+			return;
+		}
+	}
+
+	tester_rsp(BTP_SERVICE_ID_AICS, AICS_SET_GAIN, CONTROLLER_INDEX,
+		   BTP_STATUS_SUCCESS);
+}
+
+void aics_mute(void)
+{
+	LOG_DBG("AICS mute");
+
+	for (int i = 0; i < CONFIG_BT_VCP_VOL_REND_AICS_INSTANCE_COUNT; i++) {
+		int err = bt_aics_mute(included.aics[i]);
+
+		if (err) {
+			tester_rsp(BTP_SERVICE_ID_AICS, AICS_MUTE,
+				   CONTROLLER_INDEX, BTP_STATUS_FAILED);
+			return;
+		}
+	}
+
+	tester_rsp(BTP_SERVICE_ID_AICS, AICS_MUTE, CONTROLLER_INDEX,
+		   BTP_STATUS_SUCCESS);
+}
+
+void aics_unmute(void)
+{
+	LOG_DBG("AICS unmute");
+
+	for (int i = 0; i < CONFIG_BT_VCP_VOL_REND_AICS_INSTANCE_COUNT; i++) {
+		int err = bt_aics_unmute(included.aics[i]);
+
+		if (err) {
+			tester_rsp(BTP_SERVICE_ID_AICS, AICS_UNMUTE,
+				   CONTROLLER_INDEX, BTP_STATUS_FAILED);
+			return;
+		}
+	}
+
+	tester_rsp(BTP_SERVICE_ID_AICS, AICS_UNMUTE, CONTROLLER_INDEX,
+		   BTP_STATUS_SUCCESS);
+}
+
+void aics_man_gain(void)
+{
+	LOG_DBG("AICS manual gain set");
+
+	for (int i = 0; i < CONFIG_BT_VCP_VOL_REND_AICS_INSTANCE_COUNT; i++) {
+		int err = bt_aics_manual_gain_set(included.aics[i]);
+
+		if (err) {
+			tester_rsp(BTP_SERVICE_ID_AICS, AICS_MAN_GAIN,
+				   CONTROLLER_INDEX, BTP_STATUS_FAILED);
+			return;
+		}
+	}
+
+	tester_rsp(BTP_SERVICE_ID_AICS, AICS_MAN_GAIN, CONTROLLER_INDEX,
+		   BTP_STATUS_SUCCESS);
+}
+
+void aics_auto_gain(void)
+{
+	LOG_DBG("AICS auto gain set");
+	for (int i = 0; i < CONFIG_BT_VCP_VOL_REND_AICS_INSTANCE_COUNT; i++) {
+		int err = bt_aics_automatic_gain_set(included.aics[i]);
+
+		if (err) {
+			tester_rsp(BTP_SERVICE_ID_AICS, AICS_AUTO_GAIN,
+				   CONTROLLER_INDEX, BTP_STATUS_FAILED);
+			return;
+		}
+	}
+
+	tester_rsp(BTP_SERVICE_ID_AICS, AICS_AUTO_GAIN, CONTROLLER_INDEX,
+		   BTP_STATUS_SUCCESS);
+}
+
+/* TODO: Need new API function for this */
+void aics_auto_gain_only(void)
+{
+	tester_rsp(BTP_SERVICE_ID_AICS, AICS_AUTO_GAIN_ONLY, CONTROLLER_INDEX,
+		   BTP_STATUS_FAILED);
+}
+
+/* TODO: Need new API function for this */
+void aics_auto_man_only(void)
+{
+	tester_rsp(BTP_SERVICE_ID_AICS, AICS_MAN_GAIN_ONLY, CONTROLLER_INDEX,
+		   BTP_STATUS_FAILED);
+}
+
+/* TODO: Need new API function for this */
+void aics_mute_disable(void)
+{
+	tester_rsp(BTP_SERVICE_ID_AICS, AICS_MUTE_DISABLE,
+		   CONTROLLER_INDEX, BTP_STATUS_FAILED);
+}
+
+void aics_desc(uint8_t *data)
+{
+	const struct aics_audio_desc *cmd = (void *) data;
+
+	LOG_DBG("AICS description");
+
+	if (cmd->desc_len > BT_AICS_MAX_INPUT_DESCRIPTION_SIZE) {
+		LOG_ERR("Too long input (%u chars required)",
+			BT_AICS_MAX_INPUT_DESCRIPTION_SIZE);
+		goto rsp;
+	}
+
+	for (int i = 0; i < CONFIG_BT_VCP_VOL_REND_AICS_INSTANCE_COUNT; i++) {
+		int err = bt_aics_description_set(included.aics[i], cmd->desc);
+
```

Comment:
`bt_aics_description_set` expects a NULL terminate string. Thus `cmd->desc` shall be NULL terminated. If `cmd->desc` is NULL terminated, we could omit the `cmd->desc_len` and just do `strlen(cmd->desc)`. 

IMO we should still consider using `char` for `cmd->desc` for that reason, and also since `bt_aics_description_set` takes a `char *` as input so that we are more "type correct" (even though `uint8_t` and `char` can be considered the same type)

Reply:
I'd prefer if we keep string in BTP as any binary data (len+value with no null termination).

If zephyr API requires null terminated string than we just use tmp value for this

Reply:
That would be fine with me

Reply:
Done